api/src/repositories/cartRepo.ts

```diff
@@ -0,0 +1,253 @@
+          [existingItem.cart_item_id],
+        );
+
+        return objectToCamelCase<CartItem>(updated as DatabaseRow);
+      } else {
+        // Insert new item
+        const now = new Date().toISOString();
+        const { sql, values } = buildInsertSQL('cart_items', {
+          cartId,
+          productId,
+          quantity,
+          addedAt: now,
+        });
+
+        const result = await this.db.run(sql, values);
+
+        const createdItem = await this.db.get<DatabaseRow>(
+          'SELECT * FROM cart_items WHERE cart_item_id = ?',
+          [result.lastID],
+        );
+
+        if (!createdItem) {
+          throw new Error('Failed to retrieve created cart item');
+        }
+
+        return objectToCamelCase<CartItem>(createdItem);
+      }
+    } catch (error) {
+      handleDatabaseError(error);
+    }
+  }
+
+  /**
+   * Update item quantity in cart
+   */
+  async updateItemQuantity(cartItemId: number, quantity: number): Promise<void> {
+    try {
+      if (quantity < 1) {
+        throw new ValidationError('Quantity must be at least 1');
+      }
+
+      const result = await this.db.run(
+        'UPDATE cart_items SET quantity = ? WHERE cart_item_id = ?',
+        [quantity, cartItemId],
+      );
+
+      if (result.changes === 0) {
+        throw new NotFoundError('CartItem', cartItemId);
+      }
+    } catch (error) {
+      handleDatabaseError(error, 'CartItem', cartItemId);
+    }
+  }
+
+  /**
+   * Remove item from cart
+   */
+  async removeItem(cartItemId: number): Promise<void> {
+    try {
+      const result = await this.db.run(
+        'DELETE FROM cart_items WHERE cart_item_id = ?',
+        [cartItemId],
+      );
+
+      if (result.changes === 0) {
+        throw new NotFoundError('CartItem', cartItemId);
+      }
+    } catch (error) {
+      handleDatabaseError(error, 'CartItem', cartItemId);
+    }
+  }
+
+  /**
+   * Clear all items from cart
+   */
+  async clearCart(cartId: number): Promise<void> {
+    try {
+      await this.db.run('DELETE FROM cart_items WHERE cart_id = ?', [cartId]);
+    } catch (error) {
+      handleDatabaseError(error);
+    }
+  }
+
+  /**
+   * Update cart's updated_at timestamp
+   */
+  async touchCart(cartId: number): Promise<void> {
+    try {
+      const now = new Date().toISOString();
+      await this.db.run(
+        'UPDATE carts SET updated_at = ? WHERE cart_id = ?',
+        [now, cartId],
+      );
+    } catch (error) {
+      handleDatabaseError(error);
+    }
+  }
+}
+
+// Factory function to create repository instance
+export async function createCartRepository(isTest: boolean = false): Promise<CartRepository> {
+  const db = await getDatabase(isTest);
+  return new CartRepository(db);
+}
+
+// Singleton instance for default usage
+let cartRepo: CartRepository | null = null;
+
+export async function getCartRepository(isTest: boolean = false): Promise<CartRepository> {
+  if (!cartRepo) {
+    cartRepo = await createCartRepository(isTest);
```

Comment:
The `getCartRepository` function is missing the test environment check that's present in other repositories. Other repositories (branches, orders, deliveries, suppliers) check for test environment with:
```typescript
const isTestEnv = isTest || process.env.NODE_ENV === 'test' || process.env.VITEST === 'true';
if (isTestEnv) {
  return createCartRepository(true);
}
```
This ensures that test runs always get a fresh repository instance bound to the in-memory database, preventing test interference. Without this, the singleton pattern could cause tests to share state across test cases.
```suggestion
  const isTestEnv = isTest || process.env.NODE_ENV === 'test' || process.env.VITEST === 'true';
  if (isTestEnv) {
    return await createCartRepository(true);
  }
  if (!cartRepo) {
    cartRepo = await createCartRepository(false);
```